tests/test_cli.py

```diff
@@ -236,35 +262,99 @@ def test_studies_command() -> None:
+                    "n_trials",
+                    "datetime_start",
+                ]
+            else:
+                expected_keys_1 = ["name", "direction_0", "n_trials", "datetime_start"]
+                expected_keys_2 = [
+                    "name",
+                    "direction_0",
+                    "direction_1",
+                    "n_trials",
+                    "datetime_start",
+                ]
+        else:
+            expected_keys_1 = ["name", "direction", "n_trials", "datetime_start"]
+            expected_keys_2 = ["name", "direction", "n_trials", "datetime_start"]
+
+        assert len(studies) == 2
+        if output_format is None or output_format == "table":
+            assert list(studies[0].keys()) == expected_keys_1
+            assert list(studies[1].keys()) == expected_keys_2
+        else:
+            assert set(studies[0].keys()) == set(expected_keys_1)
+            assert set(studies[1].keys()) == set(expected_keys_2)
+
+        # Check study_name, direction, and n_trials for the first study.
+        assert studies[0]["name"] == study_1.study_name
+        if output_format is None or output_format == "table":
+            assert studies[0]["n_trials"] == "0"
+            if flatten:
+                assert studies[0]["direction_0"] == "MINIMIZE"
+            else:
+                assert eval(studies[0]["direction"]) == ("MINIMIZE",)
+        else:
+            assert studies[0]["n_trials"] == 0
+            if flatten:
+                assert studies[0]["direction_0"] == "MINIMIZE"
+            else:
+                assert studies[0]["direction"] == [
+                    "MINIMIZE",
+                ]
 
-        # Check study_name and n_trials for the second study.
-        elms = get_row_elements(4)
-        assert elms[0] == study_2.study_name
-        assert elms[2] == "10"
+        # Check study_name, direction, and n_trials for the second study.
+        assert studies[1]["name"] == study_2.study_name
+        if output_format is None or output_format == "table":
+            assert studies[1]["n_trials"] == "10"
+            if flatten:
+                assert studies[1]["direction_0"] == "MINIMIZE"
+                assert studies[1]["direction_1"] == "MAXIMIZE"
+            else:
+                assert eval(studies[1]["direction"]) == ("MINIMIZE", "MAXIMIZE")
+        else:
+            assert studies[1]["n_trials"] == 10
+            if flatten:
+                assert studies[1]["direction_0"] == "MINIMIZE"
+                assert studies[1]["direction_1"] == "MAXIMIZE"
+            else:
+                assert studies[1]["direction"] == ["MINIMIZE", "MAXIMIZE"]
 
 
 @pytest.mark.parametrize("objective", (objective_func, objective_func_branched_search_space))
-def test_trials_command(objective: Callable[[Trial], float]) -> None:
+@pytest.mark.parametrize("output_format", (None, "table", "json", "yaml"))
+@pytest.mark.parametrize("flatten", (True, False))
```

Comment:
How about splitting the function into two depending on the value of `flatten`? All the assertion is in the conditional branch of the `flatten` value. If possible, I'd like to avoid the [conditional test logic](http://xunitpatterns.com/Conditional%20Test%20Logic.html), which makes test code complicated.

Reply:
Thank you for pointing it out! I split flatten and nested tests.